src/AspNet.Security.OAuth.Douyin/DouyinAuthenticationHandler.cs

```diff
@@ -0,0 +1,235 @@
+        [NotNull] ClaimsIdentity identity,
+        [NotNull] AuthenticationProperties properties,
+        [NotNull] OAuthTokenResponse tokens)
+    {
+        // See https://developer.open-douyin.com/docs/resource/zh-CN/dop/develop/openapi/account-permission/get-account-open-info for details.
+        var parameters = new SortedDictionary<string, string?>()
+        {
+            ["open_id"] = tokens.Response!.RootElement.GetProperty("open_id").GetString()!,
+            ["access_token"] = tokens.AccessToken,
+        };
+
+        using var userInfoRequestContent = new FormUrlEncodedContent(parameters);
+
+        using var response = await Backchannel.PostAsync(Options.UserInformationEndpoint, userInfoRequestContent, Context.RequestAborted);
+
+        if (!response.IsSuccessStatusCode)
+        {
+            await Log.UserProfileErrorAsync(Logger, response, Context.RequestAborted);
+            throw new HttpRequestException("An error occurred while retrieving user information.");
+        }
+
+        using var stream = await response.Content.ReadAsStreamAsync(Context.RequestAborted);
+        using var document = await JsonDocument.ParseAsync(stream);
+        var mainElement = document.RootElement.GetProperty("data");
+
+        if (!ValidateReturnCode(mainElement, out var errorCode))
+        {
+            throw new AuthenticationFailureException($"An error (ErrorCode:{errorCode}) occurred while retrieving user information.");
+        }
+
+        identity.AddClaim(new Claim(ClaimTypes.NameIdentifier, mainElement.GetString("open_id")!, ClaimValueTypes.String, Options.ClaimsIssuer));
+
+        var principal = new ClaimsPrincipal(identity);
+        var context = new OAuthCreatingTicketContext(principal, properties, Context, Scheme, Options, Backchannel, tokens, mainElement);
+
+        context.RunClaimActions();
+
+        await Events.CreatingTicket(context);
+
+        return new AuthenticationTicket(context.Principal!, context.Properties, Scheme.Name);
+    }
+
+    protected override string FormatScope([NotNull] IEnumerable<string> scopes) => string.Join(',', Options.Scope);
+
+    /// <summary>
+    /// Check the code sent back by server for potential server errors.
+    /// </summary>
+    /// <param name="element">Main part of json document from response</param>
+    /// <param name="errorCode">Returned error_code from server</param>
+    /// <remarks>See https://developer.open-douyin.com/docs/resource/zh-CN/dop/develop/openapi/status-code for details.</remarks>
+    /// <returns>True if succeed, otherwise false.</returns>
+    private static bool ValidateReturnCode(JsonElement element, out int errorCode)
+    {
+        if (!element.TryGetProperty("error_code", out JsonElement errorCodeElement))
+        {
+            errorCode = 0;
+            return true;
+        }
+
+        errorCode = errorCodeElement.GetInt32()!;
+        return errorCode == 0;
```

Comment:
Just to only have one `return`.

```suggestion
        errorCode = 0;

        if (element.TryGetProperty("error_code", out JsonElement errorCodeElement))
        {
            errorCode = errorCodeElement.GetInt32()!;
        }

        return errorCode == 0;
```